cmd/minishift/cmd/config/config.go

```diff
@@ -48,98 +50,98 @@ var settingsList []Setting
+	WarnCheckNetworkHost  = createConfigSetting("warn-check-network-host", SetBool, nil, nil, true, nil)
+	SkipCheckNetworkPing  = createConfigSetting("skip-check-network-ping", SetBool, nil, nil, true, nil)
+	WarnCheckNetworkPing  = createConfigSetting("warn-check-network-ping", SetBool, nil, nil, true, nil)
+	SkipCheckNetworkHTTP  = createConfigSetting("skip-check-network-http", SetBool, nil, nil, true, nil)
+	WarnCheckNetworkHTTP  = createConfigSetting("warn-check-network-http", SetBool, nil, nil, true, nil)
+	SkipCheckStorageMount = createConfigSetting("skip-check-storage-mount", SetBool, nil, nil, true, nil)
+	WarnCheckStorageMount = createConfigSetting("warn-check-storage-mount", SetBool, nil, nil, true, nil)
+	SkipCheckStorageUsage = createConfigSetting("skip-check-storage-usage", SetBool, nil, nil, true, nil)
+	WarnCheckStorageUsage = createConfigSetting("warn-check-storage-usage", SetBool, nil, nil, true, nil)
 
 	// Pre-flight values
-	CheckNetworkHttpHost = createConfigSetting("check-network-http-host", SetString, nil, nil, true)
-	CheckNetworkPingHost = createConfigSetting("check-network-ping-host", SetString, nil, nil, true)
+	CheckNetworkHttpHost = createConfigSetting("check-network-http-host", SetString, nil, nil, true, "http://minishift.io/index.html")
+	CheckNetworkPingHost = createConfigSetting("check-network-ping-host", SetString, nil, nil, true, "8.8.8.8")
 
 	// Network settings (Hyper-V only)
-	NetworkDevice = createConfigSetting("network-device", SetString, nil, nil, true)
-	IPAddress     = createConfigSetting("network-ipaddress", SetString, []setFn{validations.IsValidIPv4Address}, nil, true)
-	Netmask       = createConfigSetting("network-netmask", SetString, []setFn{validations.IsValidNetmask}, nil, true)
-	Gateway       = createConfigSetting("network-gateway", SetString, []setFn{validations.IsValidIPv4Address}, nil, true)
-	NameServer    = createConfigSetting("network-nameserver", SetString, []setFn{validations.IsValidIPv4Address}, nil, true)
+	NetworkDevice = createConfigSetting("network-device", SetString, nil, nil, true, nil)
+	IPAddress     = createConfigSetting("network-ipaddress", SetString, []setFn{validations.IsValidIPv4Address}, nil, true, nil)
+	Netmask       = createConfigSetting("network-netmask", SetString, []setFn{validations.IsValidNetmask}, nil, true, nil)
+	Gateway       = createConfigSetting("network-gateway", SetString, []setFn{validations.IsValidIPv4Address}, nil, true, nil)
+	NameServer    = createConfigSetting("network-nameserver", SetString, []setFn{validations.IsValidIPv4Address}, nil, true, nil)
 )
 
-func createConfigSetting(name string, set func(MinishiftConfig, string, string) error, validations []setFn, callbacks []setFn, isApply bool) *Setting {
+func createConfigSetting(name string, set func(MinishiftConfig, string, string) error, validations []setFn, callbacks []setFn, isApply bool, defaultVal interface{}) *Setting {
```

Comment:
right... `interface` was needed, as a default value can either be an integer, string or boolean

Reply:
Yes, it came when @budhrg pointed it out at first. Thanks @budhrg  :)